app/ModelFunctions/LDAPFunctions.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+namespace App\ModelFunctions;
+
+use App\Exceptions\LDAPException;
+use App\Models\Configs;
+use App\Models\Logs;
+
+class LDAPFunctions
+{
+	/* @var resource $con holds the LDAP connection */
+	protected $con = null;
+
+	/* @var int $bound What type of connection does already exist? */
+	protected $bound = 0; // 0: anonymous, 1: user, 2: superuser
```

Comment:
```suggestion
	protected int $bound = 0; // 0: anonymous, 1: user, 2: superuser
```

Please use attributes with proper types. This remark is just an example and holds for all similar code lines.

Reply:
The type of $con will be dependent of the definition of ldap_connect. So I think it is more robust to not fix the type.
But when the type is an internal choice then I agree that it should be fixed whenever feasable.